lib/ansible/modules/system/user.py

```diff
@@ -1813,6 +1866,24 @@ def modify_user_usermod(self):
 
         return (rc, out, err)
 
+    def user_info(self):
+        info = super(SunOS, self).user_info()
+        if info:
+            info += self._user_attr_info()
+        return info
+
+    def _user_attr_info(self):
+        info = [''] * 3
+        with open(self.USER_ATTR, 'rb') as file_handler:
```

Comment:
After looking at this some more, I don't see a good reason to open the file in binary mode only to convert each line to native string type for further manipulation.
```suggestion
        with open(self.USER_ATTR, 'r') as file_handler:
```

Reply:
Updated this.